UmpleToPhp/UmpleTLTemplates/class_MethodDeclaration.ump

```diff
@@ -72,7 +74,130 @@ class UmpleToPhp {
   			{
     			appendln(realSb, "\n  {");
     			if (customPreconditionCode != null) { append(realSb, "\n{0}\n",GeneratorHelper.doIndent(customPreconditionCode, "    "));}
+    			if (customBeforeInjectionCode != null) { append(realSb, "\n{0}\n",GeneratorHelper.doIndent(customBeforeInjectionCode, "    "));}
+
+          if(properMethodBody.contains("return"))
+          {
+            if(customAfterInjectionCode != null) {
+              // Do some pre-processing to handle returns not being on a new line. Doing this allows us to maintain suitable indentation.
+              String[] properMethodLines = properMethodBody.split("\\n");
+              String fixedProperMethodBody = "";
+              for(int i = 0; i < properMethodLines.length; i++) {
```

Comment:
Use a for each here, especially since you index into the array quite a few times.